x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/CrossClusterAccessAuthenticationService.java

```diff
@@ -108,6 +111,28 @@ public void authenticate(final String action, final TransportRequest request, fi
         }
     }
 
+    public void tryAuthenticateCredentialsHeaderOnly(
+        ThreadContext threadContext,
+        Map<String, String> headers,
+        ActionListener<AuthenticationResult<User>> listener
```

Comment:
Can this parameter be changed to `ActionListener<Void> listener`?
The idea is to expose only a binary option to the caller, either to let the request through or to reject it.

Reply:
Indeed -- I did this separately without actually seeing your comment 😅